### PR TITLE
feat: support predicate pushdown for parquet files

### LINKED ISSUE #1540: Support predicate pushdown for parquet file
As part of Predicate pushdown feature #734, the work for RUB has been done in #1527. This ticket is to track the similar work but for parquet file.

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #1540 

Add tests and through debug discovered the appropriate predicates are pushed down for parquet files

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: server/src/query_tests/scenarios.rs [hunk 1]
```diff
@@ -704,9 +704,31 @@ pub(crate) async fn make_one_rub_chunk_scenario(
             .unwrap();
     }
     let scenario1 = DbScenario {
-        scenario_name: "Data in read buffer".into(),
+        scenario_name: "--------------------- Data in read buffer".into(),
+        db,
+    };
+
+    // Scenario 2: One closed chunk in Parquet only
```
Comment (NGA-TRAN) [inline server/src/query_tests/scenarios.rs:711]: By adding this scenario, all SQLs in [sql_predicate_pushdown](https://github.com/influxdata/influxdb_iox/blob/9dfc653bac5181e91e1ec42b49e1c1d9753c7fdf/server/src/query_tests/sql.rs#L414) will be run for on parquet files, too

File: server/src/query_tests/sql.rs [hunk 2]
```diff
@@ -641,4 +641,37 @@ async fn sql_predicate_pushdown() {
         "SELECT * from restaurant where 5.0 < system and town != 'tewsbury' and system < 7.0 and (count = 632 or town = 'reading') and time > to_timestamp('1970-01-01T00:00:00.000000130+00:00')",
         &expected
     );
+
+    // TODO: Hit stackoverflow in DF. Ticket https://github.com/apache/arrow-datafusion/issues/419
```
Comment (NGA-TRAN) [inline server/src/query_tests/sql.rs:645]: My curiosity to add one more equal expression lead to this stack overflow discovery

File: server/src/query_tests/sql.rs [hunk 3]
```diff
@@ -641,4 +641,37 @@ async fn sql_predicate_pushdown() {
...
+    //     "| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading |",
+    //     "+-------+--------+-------------------------------+---------+",
+    // ];
+    // run_sql_test_case!(
+    //     TwoMeasurementsPredicatePushDown {},
+    //     "SELECT * from restaurant where system > 5.0 and 'tewsbury' != town and system < 7.0 and town = 'reading'",
+    //     &expected
+    // );
+
+    // Test 13: three push-down expression: system > 5.0 and system < 7.0 and town = 'reading'
```
Comment (NGA-TRAN) [inline server/src/query_tests/sql.rs:662]: To avoid hiring stack overflow, I reduce the non-eq expression to test the equal one

**Comment by NGA-TRAN** [comment]
My next step is to work on [#1538](https://github.com/influxdata/influxdb_iox/issues/1538) to have predicate pushdown shown in the explain plan to avoid the needs for debugging to see whether the predicate is pushed down or not

File: server/src/query_tests/scenarios.rs (hunk 1 above)
Comment (alamb) [reply server/src/query_tests/scenarios.rs:711]: so cool!

**Review by alamb — APPROVED** [review]
This is very cool. Nice work @NGA-TRAN 

I also tried this test and I got a 
```

thread 'IOx Executor Thread' has overflowed its stack
fatal runtime error: stack overflow
error: test failed, to rerun pass '-p server --lib'
```

I'll try and debug it a little more

**Pull request merged**